android/src/main/java/flutter/plugins/contactsservice/contactsservice/ContactsServicePlugin.java

```diff
@@ -537,6 +552,55 @@ private Cursor getCursorForPhone(String phone) {
     return null;
   }
 
+  private ArrayList<Contact> getContactsForEmailOrName(String query, boolean localizedLabels) {
+    if (query.isEmpty())
+      return null;
+    Uri uri = Uri.withAppendedPath(Email.CONTENT_FILTER_URI, Uri.encode(query));
+    Cursor cursor = contentResolver.query(uri, PROJECTION_EMAIL_NAME, null, null, null);
+    return getContactsForEmailOrNameFrom(cursor, localizedLabels);
+  }
+
+  /**
+   * Builds the list of contacts from the cursor for email or name only
+   * @param cursor email cursor filtered
+   * @return the list of contacts
+   */
+  private ArrayList<Contact> getContactsForEmailOrNameFrom(Cursor cursor, boolean localizedLabels) {
+    HashMap<String, Contact> map = new LinkedHashMap<>();
+
+    while (cursor != null && cursor.moveToNext()) {
+      try {
+        String contactId = cursor.getString(cursor.getColumnIndexOrThrow(ContactsContract.Data.CONTACT_ID));
+        if (!map.containsKey(contactId)) {
+          map.put(contactId, new Contact(contactId));
+        }
+        Contact contact = map.get(contactId);
+
+        String mimeType = cursor.getString(cursor.getColumnIndexOrThrow(ContactsContract.Data.MIMETYPE));
+
+        // NAME
+        contact.displayName = cursor.getString(cursor.getColumnIndexOrThrow(ContactsContract.Contacts.DISPLAY_NAME));
+
+        //MAILS
+        if (mimeType.equals(CommonDataKinds.Email.CONTENT_ITEM_TYPE)) {
+          String email = cursor.getString(cursor.getColumnIndexOrThrow(Email.ADDRESS));
+          int type = cursor.getInt(cursor.getColumnIndexOrThrow(Email.TYPE));
+          if (!TextUtils.isEmpty(email)) {
+            String label = Item.getEmailLabel(resources, type, cursor, localizedLabels);
+            contact.emails.add(new Item(label, email, type));
+          }
+        }
+      } catch (java.lang.Exception e) {
+        e.printStackTrace();
+      }
+    }
+
+    if(cursor != null)
+      cursor.close();
+
```

Comment:
```suggestion
    if (cursor != null)
      cursor.close();

```

Reply:
This code is from library, i didn't change anything. So i think we don't have to fix invention error